commons/src/main/java/org/infinispan/commons/marshall/WrappedByteArray.java

```diff
@@ -74,7 +73,7 @@ public boolean equalsWrappedBytes(WrappedBytes other) {
 
    @Override
    public int hashCode() {
-      return hashCode;
+      return hashCode == 0 ? (hashCode = Arrays.hashCode(bytes)) : hashCode;
```

Comment:
I guess it's used for reading as well, so a write operation would benefit from the cached hash code.

OTOH with compressed oops removing the field will remove 8 bytes from `WrappedByteArray`, and keys aren't supposed to be that long anyway, so I would still remove the field.